lib/pos_blockchain.go

```diff
@@ -0,0 +1,451 @@
+package lib
+
+import (
+	"github.com/deso-protocol/core/collections"
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+	"time"
+)
+
+// processBlockPoS runs the Fast-Hotstuff block connect and commit rule as follows:
+//  1. Determine if we're missing a parent block of this block and any of its parents from the block index.
+//     If so, return the hash of the missing block and add this block to the orphans list.
+//  2. Validate on an incoming block, its header, its block height, the leader, and its QCs (vote or timeout)
+//  3. Store the block in the block index and uncommitted blocks map.
+//  4. Resolves forks within the last two blocks
+//  5. Connect the block to the blockchain's tip
+//  6. Run the commit rule - If applicable, flushes the incoming block's grandparent to the DB
+//  7. Prune in-memory struct holding uncommitted block.
+//  8. Update the currentView to this new block's view + 1
+func (bc *Blockchain) processBlockPoS(desoBlock *MsgDeSoBlock, currentView uint64, verifySignatures bool) (_success bool, _isOrphan bool, _missingBlockHashes []*BlockHash, _err error) {
```

Comment:
@lazynina this looks incomplete. Lmk if you want me to review it now or wait?

Reply:
Marking this as resolved since it's implemented in https://github.com/deso-protocol/core/pull/742